spring-integration-rsocket/src/main/java/org/springframework/integration/rsocket/outbound/RSocketOutboundGateway.java

```diff
@@ -0,0 +1,303 @@
+		setAsync(true);
+		setPrimaryExpression(this.routeExpression);
+	}
+
+	public void setDataMimeType(MimeType dataMimeType) {
+		Assert.notNull(dataMimeType, "'dataMimeType' must not be null");
+		this.dataMimeType = dataMimeType;
+	}
+
+	public void setFactoryConfigurer(@Nullable Consumer<RSocketFactory.ClientRSocketFactory> factoryConfigurer) {
+		this.factoryConfigurer = factoryConfigurer;
+	}
+
+	public void setStrategiesConfigurer(@Nullable Consumer<RSocketStrategies.Builder> strategiesConfigurer) {
+		this.strategiesConfigurer = strategiesConfigurer;
+	}
+
+	public void setCommand(Command command) {
+		setCommandExpression(new ValueExpression<>(command));
+	}
+
+	public void setCommandExpression(Expression commandExpression) {
+		Assert.notNull(commandExpression, "'commandExpression' must not be null");
+		this.commandExpression = commandExpression;
+	}
+
+	/**
+	 * Configure a type for a request {@link Publisher} elements.
+	 * @param publisherElementType the type of the request {@link Publisher} elements.
+	 * @see RSocketRequester.RequestSpec#data(Publisher, Class)
+	 */
+	public void setPublisherElementType(Class<?> publisherElementType) {
+		Assert.notNull(publisherElementType, "'publisherElementType' must not be null");
+		setPublisherElementTypeExpression(new ValueExpression<>(publisherElementType));
+
+	}
+
+	/**
+	 * Configure a SpEL expression to evaluate a request {@link Publisher} elements type at runtime against
+	 * a request message.
+	 * @param publisherElementTypeExpression the expression to evaluate a type for the request
+	 * {@link Publisher} elements.
+	 * @see RSocketRequester.RequestSpec#data
+	 */
+	public void setPublisherElementTypeExpression(Expression publisherElementTypeExpression) {
+		this.publisherElementTypeExpression = publisherElementTypeExpression;
+	}
+
+	/**
+	 * Specify the expected response type for the RSocket response.
+	 * @param expectedResponseType The expected type.
+	 * @see #setExpectedResponseTypeExpression(Expression)
+	 * @see RSocketRequester.ResponseSpec#retrieveMono
+	 * @see RSocketRequester.ResponseSpec#retrieveFlux
+	 */
+	public void setExpectedResponseType(Class<?> expectedResponseType) {
+		Assert.notNull(expectedResponseType, "'expectedResponseType' must not be null");
+		setExpectedResponseTypeExpression(new ValueExpression<>(expectedResponseType));
+	}
+
+	/**
+	 * Specify the {@link Expression} to determine the type for the RSocket response.
+	 * @param expectedResponseTypeExpression The expected response type expression.
+	 * @see RSocketRequester.ResponseSpec#retrieveMono
+	 * @see RSocketRequester.ResponseSpec#retrieveFlux
+	 */
+	public void setExpectedResponseTypeExpression(Expression expectedResponseTypeExpression) {
+		this.expectedResponseTypeExpression = expectedResponseTypeExpression;
+	}
+
+
+	@Override
+	protected void doInit() {
+		super.doInit();
+		RSocketRequester.Builder builder = RSocketRequester.builder();
+		if (this.factoryConfigurer != null) {
```

Comment:
It's probably cleaner to default to a no-op configurers and assert not null in setters.